src/main/java/mate/academy/rickandmorty/service/CharacterClient.java

```diff
@@ -0,0 +1,68 @@
+package mate.academy.rickandmorty.service;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.io.IOException;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.util.ArrayList;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import mate.academy.rickandmorty.dto.external.ApiResponseData;
+import mate.academy.rickandmorty.dto.external.CharacterResponseDto;
+import org.springframework.stereotype.Component;
+
+@Component
+@RequiredArgsConstructor
+public class CharacterClient {
+
+    private static final String URL_CHARACTERS = "https://rickandmortyapi.com/api/character";
+
+    private final ObjectMapper objectMapper;
+
+    public List<CharacterResponseDto> getCharacters() {
+
+        HttpClient httpClient = HttpClient.newHttpClient();
```

Comment:
Why http client should be instantiated each time method is invoked

Make it bean.
```suggestion
        HttpClient httpClient = HttpClient.newHttpClient();
```

Reply:
consider using RestTemplate to avoid using object mapper